ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionIndexing.scala

```diff
@@ -141,16 +108,17 @@ object TransactionIndexing {
         offset: Offset,
         divulgedContracts: Iterable[DivulgedContract],
     ): TransactionIndexing = {
-      val created = creates.result()
-      val archived = archives.result()
-      val allCreatedContractIds = created.map(_.coid)
-      val allContractIds = allCreatedContractIds.union(archived)
-      val netCreates = created.filterNot(c => archived(c.coid))
-      val netArchives = archived.filterNot(allCreatedContractIds)
-      val netDivulgedContracts = divulgedContracts.filterNot(c => allContractIds(c.contractId))
-      val netTransactionVisibility =
-        Relation.from(visibility.result()).view.filterKeys(!archived(_)).toMap
-      val netVisibility = Relation.union(netTransactionVisibility, visibility(netDivulgedContracts))
+      val divulgedContractIndex = divulgedContracts
+        .map(divulgedContract => divulgedContract.contractId -> divulgedContract)
+        .toMap
+      val netDivulged = blinding.divulgence.map {
```

Comment:
Why `netDivulged`? It's not clear to me what kind of netting we are doing here.

Suggestion: rename to `divulgenceEvents`

Reply:
At this point they are more divulged contracts, as this is the indexing part...or would it be cleaner jut to drop ContractsInfo entirely and add this to the TransactionInfo, under the name divulgedEvents? That might be even more cleaner :)